docs/build/guides/conventions/stellar-asset-contract.mdx

```diff
@@ -0,0 +1,94 @@
+---
+title: Integrate Stellar Assets Contracts
+description: Test and use Stellar assets in a Soroban smart contract
+---
+
+From the contract perspective Stellar Asset Contract is not different from any other token that implements the Soroban [SEP-41 Token Interface]. The Rust SDK contains a pregenerated client for any contract that implements the token interface:
+
+```rust
+use soroban_sdk::token;
+
+struct MyContract;
+
+#[contractimpl]
+impl MyContract {
+  fn token_fn(e: Env, id: Address) {
+    // Create a client instance for the provided token identifier. If the id
+    // value corresponds to an SAC contract, then SAC implementation is used.
+    let client = token::TokenClient::new(&env, &id);
+    // Call token operations part of the SEP-41 token interface
+    client.transfer(...);
+  }
+}
+```
+
+The address parameter is not the address of the issuer of an asset, it corresponds to the deployed contract address for this asset.
+
+```bash
+stellar contract id asset \
+    --source G... \
+    --network testnet \
+    --asset [asset:issuer]
+```
+
+E.g. for USDC, it would be `--asset USDC:G...` For the native asset, XLM, `--asset native`. See the [deploy SAC] guide for more details.
+
+[deploy SAC]: ../cli/deploy-stellar-asset-contract.mdx
+
+:::info Clients
+
+A client created by [`token::Client`] implements the functions defined by any contract that implements the [SEP-41 Token Interface]. But with [CAP-46-6 smart contract standardized asset], the Stellar Asset Contract exposes additional functions such as `mint`. To access the additional functions, another client needs to be used: [`token::StellarAssetClient`]. This client only implements the functions from CAP-46-6, which are not part of the SEP-41.
+
+```rust
+let client = token::StellarAssetClient::new(&env, &id);
+// Call token operations which are not part of the SEP-41 token interface
+// but part of the CAP-46-6 Smart Contract Standardized Asset
+client.mint(...);
+```
+
+:::
+
+### Testing
+
+Soroban Rust SDK provides an easy way to instantiate a Stellar Asset Contract tokens using `register_stellar_asset_contract`. This function can be seen as the deployment of a generic token. In the following example, we are following the best practices outlined in the [Issuing and Distribution Accounts section](../../../tokens/control-asset-access.mdx#issuing-and-distribution-accounts):
+
+```rust
+#![cfg(test)]
+
+use soroban_sdk::testutils::Address as _;
+use soroban_sdk::{token, Address, Env};
+use token::TokenClient;
+use token::StellarAssetClient;
+
+#[test]
+fn test() {
+    let e = Env::default();
+    e.mock_all_auths();
+
+    let issuer = Address::random();
+    let distributer = Address::random();
+
+    let token_address = e.register_stellar_asset_contract(issuer.clone());
+
+    // client for SEP-41 operations
+    let token = TokenClient::new(&e, &token_address);
+    // client for Stellar Asset Contract operations
```

Comment:
I think using the word `functions` instead of `operations` _might_ make more sense in these explainer comments. I can see how, when we're using SAC functions, it's analogous to the "classic" tx operations, so it makes sense to me. But, if someone is unfamiliar with those ops, it might not.

What I'm **sure of** is that it's splitting hairs. I would absolutely not die on this hill 🤣

Reply:
Makes sense, here we are in a dev context so functions seems better 👍